dotnet-coverage-test-data-import-guide.md

```diff
@@ -1,117 +1,137 @@
-2. Build your project using MSBuild
-3. Run your test tool, instructing it to produce a report at the same location specified earlier to the MSBuild SonarQube Runner (How to generate reports with different tools)
-4. Run the SonarScanner.MSBuild.exe `end` command
-
-#### Projects being considered as test projects by the Scanner for MSBuild
-
-Read the [ProjectCategorisation](https://github.com/SonarSource/sonar-scanner-msbuild/wiki/Analysis-of-product-projects-vs.-test-projects#project-categorisation) section to better understand if the project not having code coverage may be considered a test project.
-
-#### Not running all commands from the same folder
-
-All commands, including the code coverage generation, must be run from the same folder.
-
-To check which is the base path look for this log:
-
-```
-DEBUG: The current user dir is '<PATH>'.
-```
-
-To check the path of the indexed files look for this log:
-
-```
-DEBUG: '<PATH\FILE.cs>' indexed with language 'cs'
-```
-
-Then if the paths inside the coverage file differ from the path of the indexed files, look for this log:
-
-```
-INFO: Parsing the OpenCover report <FILE PATH>
-DEBUG: Skipping file with path '<PATH\FILE.cs>' because it is not indexed or does not have the supported language.
-```
+Understand the basics of...
+* SonarScanner for MSBuild by reading its docs [here](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/), in case you are using it in your pipeline.
+* Azure Devops - understand the integration between Azure DevOps and SonarCloud by reading [this guideline](https://azuredevopslabs.com//labs/vstsextend/sonarcloud/). =====> will add the reference to the yaml file for “Prepare SonarCloudAnalysis” step......
+
+Note that SonarScanner for MSBuild has been renamed as SonarScanner for .NET.
+
+# Troubleshooting steps
+
+* Are you using a valid version of your code coverage tool?
+
+* Are the reports in the right format? (need to check which file type matches which code coverage tool)
+
+* Do the reports even exist?
+ If you see the following, then ensure the report generation for your code coverage tool actually produced a file and has content in it.
+ 
+     ```
+     WARN: The Code Coverage report doesn’t contain any coverage data for the included files.
+     ```
+ 
+    * Verify Indexed Paths Compared to Coverage File Paths
+ (see https://community.sonarsource.com/t/code-coverage-is-not-shown-in-the-sonarcloud-io/15218/6)
+* Do the reports contain valid information for the expected set of files? (give examples)
+
+* Did you specify the correct report path property to import the report at the right step?
+When importing code coverage reports, you need to ensure that you specify the report path with your associated code coverage tool's property for the SonarScanner.MSBuild.exe `begin` command, as detailed in [Test Coverage](https://docs.sonarqube.org/latest/analysis/coverage/):
```

Comment:
We also have a automatic detection and write of 
- sonar.cs.vscoveragexml.reportsPaths
- sonar.cs.vstest.reportsPaths

properties inside the Scanner for .NET, if proper trx / coverage files have been found in specific folders.

Reply:
@mickael-caro-sonarsource: Which folders are they? I can add that to make it more clear.

Reply:
IIRC, we look at $(Agent.BuildDirectory)\TestResults first, then we fallback in $(Agent.TempDirectory). This is working only on a Azure DevOps environment.

Reply:
Good call! I see it in the [Code Coverage](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/#header-5) section. I already added in the introduction section, but I'll reference it again here as well.